### PR TITLE
eager flush results for single non aggregate query

### PR DESCRIPTION
**Pull request by shz117**
this will reduce TTFB for non-agg queries from 10+s to milliseconds.

### CODE REVIEW

File: query/aql_compiler.go [hunk 1]
```diff
@@ -79,8 +80,8 @@ const (
 
 // Compile returns the compiled AQLQueryContext for data feeding and query
 // execution. Caller should check for AQLQueryContext.Error.
-func (q *AQLQuery) Compile(store memstore.MemStore, returnHLL bool) *AQLQueryContext {
-	qc := &AQLQueryContext{Query: q, ReturnHLLData: returnHLL}
+func (q *AQLQuery) Compile(store memstore.MemStore, returnHLL bool, w http.ResponseWriter) *AQLQueryContext {
```
Comment (voyager-dw) [inline query/aql_compiler.go]: it is quite strange compile needs the http writer as parameter, maybe outside assign to aqlquerycontext?

File: api/query_handler.go [hunk 2]
```diff
@@ -138,37 +137,88 @@ func (handler *QueryHandler) handleAQLInternal(aqlRequest AQLRequest, w http.Res
...
+			err = qc.Error
+			utils.GetQueryLogger().With(
+				"error", err,
+				"query", aqlQuery,
+				"context", qc,
+			).Error("Error happened when processing query")
+			statusCode = http.StatusInternalServerError
+			return
+		}
+		w.Write([]byte(`]}]`))
```
Comment (voyager-dw) [inline api/query_handler.go]: where is the [{[, inside processquery?

Comment (shz117) [reply api/query_handler.go]: yes..

File: query/aql_compiler.go (hunk 1 above)
Comment (shz117) [reply query/aql_compiler.go]: good point, I can change it

**Pull request merged**